filer/admin/patched/admin_utils.py

```diff
@@ -43,31 +48,103 @@ def format_callback(obj):
         has_admin = obj.__class__ in admin_site._registry
         opts = obj._meta
 
+        no_edit_link = '%s: %s' % (capfirst(opts.verbose_name),
+                                   force_text(obj))
+
         if has_admin:
-            admin_url = reverse('%s:%s_%s_change'
-            % (admin_site.name,
-               opts.app_label,
-               opts.object_name.lower()),
-                None, (quote(obj._get_pk_val()),))
-            p = get_delete_permission(opts)
+            try:
+                admin_url = reverse('%s:%s_%s_change'
+                                    % (admin_site.name,
+                                       opts.app_label,
+                                       opts.model_name),
+                                    None, (quote(obj._get_pk_val()),))
+            except NoReverseMatch:
+                # Change url doesn't exist -- don't display link to edit
+                return no_edit_link
+
+            p = '%s.%s' % (opts.app_label,
+                           get_permission_codename('delete', opts))
             if not user.has_perm(p):
                 perms_needed.add(opts.verbose_name)
-                # Display a link to the admin page.
-            return mark_safe('%s: <a href="%s">%s</a>' %
-                             (escape(capfirst(opts.verbose_name)),
-                              admin_url,
-                              escape(obj)))
+            # Display a link to the admin page.
+            return format_html('{}: <a href="{}">{}</a>',
+                               capfirst(opts.verbose_name),
+                               admin_url,
+                               obj)
         else:
             # Don't display link to edit, because it either has no
             # admin or is edited inline.
-            return '%s: %s' % (capfirst(opts.verbose_name),
-                               force_text(obj))
+            return no_edit_link
 
     to_delete = collector.nested(format_callback)
 
     protected = [format_callback(obj) for obj in collector.protected]
-
-    return to_delete, perms_needed, protected
+    model_count = {model._meta.verbose_name_plural: len(objs) for model, objs in collector.model_objs.items()}
+
+    return to_delete, model_count, perms_needed, protected
+
+
+class NestedObjects(Collector):
```

Comment:
Because I am using a copy of the Django 1.9 implementation of get_deleted_objects, and that only works with the matching 1.9 version of NestedObjects. When we drop Django<1.8 support this can probably be removed again.